### PR TITLE
Parameter and Permission/Security Validation

### PR DESCRIPTION
**Pull request by Teagan42**
Completely reworking to be a router.

`registerRoute` will now accept a Swagger JS OperationSchema (http://swagger.io/specification/#operationObject).

The only modification to the Swagger JS schema is on the ParameterSchema (http://swagger.io/specification/#operationObject) - the `type` field takes a `Joi` object for validation.

`/swagger` endpoint is created when `listen` is invoked (which prevents new routes from being registered)

Addresses #9 MVP

### CODE REVIEW

File: test/validation/parameterValidation.test.js [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+      expect(req.params.id)
+        .to.be.a('Date');
+
+      expect(req.params.id.toString())
+        // I hate that month is 0-index
+        .to.equal(new Date(1943, 3, 16).toString());
+    });
+
+    // it('to a string', () => {
```
Comment (Teagan42) [inline test/validation/parameterValidation.test.js]: Joi does not seem to want to convert to a string - will see what I can do about it.

Comment (jblossom-denver) [inline test/validation/parameterValidation.test.js]: Not sure, but we might not be able to chain to the constructor on instantiation like that.
might have to do it like this: 
`
var x = new Date();
myVar = x.toString();
`

Comment (Teagan42) [inline test/validation/parameterValidation.test.js]: Like what?

The issue is `Joi.string()` will not convert `42` to `'42'` - if you do `Joi.number()` then `'42'` converts to `42` just fine.

File: src/router.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    }
+
+    if (handler && typeof handler === 'function') {
+      // Actual route handler
+      handlers.push(handler);
+    }
+    else {
+      // Default handler - not sure if we want this
+      handlers.push((res, req, next) => {
+        res.status(200).send();
```
Comment (james-gibson) [inline src/router.js]: A `501 Not Implemented` might make the most sense at this point, keeps it pretty opinion free

Comment (Teagan42) [inline src/router.js]: Sounds good

**Pull request merged**